cmd/gaze/main.go

```diff
@@ -18,21 +18,53 @@ const (
 type argument struct {
 	height uint16
 	width  uint16
+	help   bool
 }
 
 // parses and return command line arguments
 func parseArguments() *argument {
 	var height = flag.Uint("h", uint(defaultHeight), fmt.Sprintf("maze height, e.g. -h %d", defaultHeight))
 	var width = flag.Uint("w", uint(defaultWidth), fmt.Sprintf("maze width, e.g. -w %d", defaultWidth))
+	var help = flag.Bool("help", false, "This is a Help Subcommand")
 
 	flag.Parse()
 
-	return &argument{height: uint16(*height), width: uint16(*width)}
+	return &argument{height: uint16(*height), width: uint16(*width), help: bool(*help)}
 }
 
 func main() {
 	arg := parseArguments()
-	k := algos.NewPrim(arg.height, arg.width)
-	k.Generate()
-	k.Board.Write(os.Stdout)
+	if arg.help == true {
+		fmt.Println(helpReturn())
```

Comment:
we could get rid of the `else` if we exit here with `os.Exit(0)`

Reply:
Done